### PR TITLE
Feature/join friend game

### PR DESCRIPTION
**Pull request by ArnaudPoletto**

### CODE REVIEW

File: app/src/main/java/sdp/moneyrun/Helpers.java [hunk 1]
```diff
@@ -154,4 +159,74 @@ public static void putPlayersInIntent(Intent intent, List<Player>players){
...
+     * @param gameRepresentation the game to join
+     * @param databaseReference the database reference
+     * @param activity the activity
+     * @param currentUser the user that joins the game
+     */
+    public static void joinLobbyFromJoinButton(@NonNull GameRepresentation gameRepresentation,
+                                               @NonNull DatabaseReference databaseReference,
+                                               @NonNull Activity activity,
+                                               @NonNull User currentUser) {
+        if(gameRepresentation.getGameId() == null){
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/Helpers.java:192]: Should be tested

File: app/src/main/java/sdp/moneyrun/database/GameDatabaseProxy.java [hunk 2]
```diff
@@ -77,6 +78,19 @@ public String putGame(@Nullable Game game) {
         return id;
     }
 
+    /**
+     * Remove a game to the database.
+     *
+     * @param game the game to be removed in the database
+     */
+    public void removeGame(@Nullable Game game) {
+        if (game == null) {
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/database/GameDatabaseProxy.java:87]: This should also be tested, at least the case where game is null

File: app/src/main/java/sdp/moneyrun/location/AndroidLocationService.java [hunk 3]
```diff
@@ -88,6 +104,40 @@ public LocationRepresentation getCurrentLocation() {
...
+     * @param location a location
+     * @param l another location
+     * @return the best location between 2 locations.
+     */
+    private Location getUpdatedLocation(Location location, Location l){
+        if(l == null){
+            return location;
+        }
+
+        if (location == null || l.getAccuracy() < location.getAccuracy()) {
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/location/AndroidLocationService.java:132]: This should also be tested

**Review by Alvaro2112 — CHANGES_REQUESTED** [review]
Very good work, I would just suggest adding tests where I left the comments. Also, do not forget to add documentation to your code.

File: app/src/main/java/sdp/moneyrun/game/GameBuilder.java [hunk 4]
```diff
@@ -136,22 +140,21 @@ public GameBuilder setIsVisible(boolean isVisible) {
...
+    public Game build() {
         checkBuildArguments();
+        Game game;
 
-        if (riddles == null )
-            new Game(name, host, players, maxPlayerCount, startLocation, isVisible, coins);
-
-        else{
+        if (riddles == null ) {
+            game = new Game(name, host, players, maxPlayerCount, startLocation, isVisible, coins, DEFAULT_NUM_COINS, DEFAULT_RADIUS, DEFAULT_DURATION);
```
Comment (gauthierboeshertz) [inline app/src/main/java/sdp/moneyrun/game/GameBuilder.java:148]: Why is there no setter for the three default values? They should not be default values when the host creates the game

**Review by gauthierboeshertz — CHANGES_REQUESTED** [review]

File: app/src/main/java/sdp/moneyrun/Helpers.java (hunk 1 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/Helpers.java:192]: done

File: app/src/main/java/sdp/moneyrun/database/GameDatabaseProxy.java (hunk 2 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/database/GameDatabaseProxy.java:87]: done

File: app/src/main/java/sdp/moneyrun/location/AndroidLocationService.java (hunk 3 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/location/AndroidLocationService.java:132]: done

File: app/src/main/java/sdp/moneyrun/game/GameBuilder.java (hunk 4 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/game/GameBuilder.java]: I updated the gameBuilder properly

**Review by gauthierboeshertz — APPROVED** [review]

**Review by Alvaro2112 — APPROVED** [review]
